lua/swenv/create.lua

```diff
@@ -0,0 +1,189 @@
+  local Job = require('plenary.job')
+  local dir_name = vim.fs.dirname(requirements_path)
+  local venv_path = dir_name .. '/' .. venv_dir
+  vim.notify(
+    'swenv.nvim: starting pip install at: ' .. requirements_path .. ' in venv: ' .. venv_path,
+    vim.log.levels.INFO
+  )
+  local python_app = 'python3'
+  if vim.fn.executable(python_app) ~= 1 then
+    python_app = 'python'
+  end
+  Job:new({
+    command = python_app,
+    args = { '-m', 'venv', venv_path },
+    cwd = dir_name,
+    on_exit = function(j, _)
+      vim.schedule(function()
+        if j.code ~= 0 then
+          vim.notify('swenv.nvim: ' .. vim.inspect(j._stderr_results .. j._stdout_results), vim.log.levels.ERROR)
+        else
+          local pip_path = venv_path .. '/' .. 'bin/pip'
+          local install_args = { 'install', '-r', requirements_path }
+          if requirements_path == 'pyproject.toml' then
+            install_args = { 'install', '.' }
+          end
+          Job:new({
+            command = pip_path,
+            args = install_args,
+            cwd = dir_name,
+            on_exit = function(k, _)
+              vim.schedule(function()
+                if k.code ~= 0 then
+                  vim.notify('swenv.nvim: ' .. vim.inspect(k._stderr_results), vim.log.levels.ERROR)
+                else
+
+                  local venv_name = vim.fs.basename(dir_name)
+                  swenv_set_venv(venv_path, venv_name)
+                  vim.notify('Set venv: ' .. venv_path, vim.log.levels.INFO)
+                end
+              end)
+            end,
+          }):start()
+        end
+      end)
+    end,
+  }):start()
+end
+
+--- Automatically create venv directory and use multiple method to auto install dependencies
+M.auto_create_set_python_venv = function()
+  local stop = false
+  local check_paths = {
+    {
+      path = venv_dir,
+      callback = function(path)
+        -- initial set, still want to do dependency install from others if available
+        local venv_name = vim.fs.basename(vim.fs.dirname(path))
+        swenv_set_venv(path, venv_name)
+      end,
+    },
+    {
+      path = 'pdm.lock',
+      callback = function(path)
+        pdm_sync(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'requirements.txt',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'dev-requirements.txt',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
+      end,
+    },
+    {
+      path = 'pyproject.toml',
+      callback = function(path)
+        pip_install_with_venv(path)
+        vim.g._auto_set_python_venv_parent_dir = vim.fs.dirname(path)
+        stop = true
```

Comment:
I'm not fully sure how this `stop` variable works? Are these callbacks run in parallel until one finishes?

Reply:
I see. Do I understand correctly that this does not actually create the venv but only install dependencies? Maybe the names should then be rephrased as by `auto_create...venv` I thought it would also create the venv if it didn't exist?

Also are currently the dependencies installed every time?